### PR TITLE
use nested transactions for adding Package and User objects

### PR DESCRIPTION
**Pull request by mikebonnet**
Replaces the confusing (and, it turns out, not 100% effective) INSERT...SELECT construct with nested transactions. Also makes it possible to run tests against a local PostgreSQL instance, since faitout is no more.

### CODE REVIEW

**Comment by mikebonnet** [comment]
@ralphbean @pypingou Mind taking a look?

File: datanommer.models/datanommer/models/__init__.py [hunk 1]
```diff
@@ -283,15 +297,19 @@ def get_or_create(cls, name):
...
-        # Use an INSERT ... SELECT to guarantee we don't get unique constraint
-        # violations if multiple instances of datanommer are trying to insert the same
-        # value at the same time.
-        not_exists = ~exists(select([cls.__table__.c.name]).where(cls.name == name))
-        insert = cls.__table__.insert(inline=True).\
-                 from_select([cls.__table__.c.name],
-                             select([literal(name)]).where(not_exists))
-        session.execute(insert)
-        return cls.query.filter_by(name=name).one()
+        obj = cls.query.filter_by(name=name).one_or_none()
```
Comment (pypingou) [inline datanommer.models/datanommer/models/__init__.py:300]: Note that the ``one_or_none`` method is not in every version of sqlalchemy, so we may want to set this lower version limit somewhere.

**Comment by pypingou** [comment]
One comment, otherwise looks good to me

File: datanommer.models/datanommer/models/__init__.py (hunk 1 above)
Comment (mikebonnet) [reply datanommer.models/datanommer/models/__init__.py:300]: Updated the sqlalchemy requirement to 1.0.9.

**Comment by pypingou** [comment]
:+1: thanks :)

**Comment by mikebonnet** [comment]
Can this be merged? I don't have access.

**Comment by cverna** [comment]
Done

**Pull request merged**